### DISCUSSION TITLE
Anomaly detections of patchcore have poor localisation

### DISCUSSION THREAD

**Discussion by amal-r1** [open-edge-platform/anomalib#526] [Q&A]
I am training a patchcore model on a custom dataset, and from the results I was able to observe that the localisation were not very good. Instead of pinpoints where the anomaly is, the model is giving a patch of area around it as anomalous. I tried a lot of things, like tuning parameters and improving of the data etc, but still the issue is persisting. The only difference between my data and the ones in mvtec is that the object I am trying to apply anomaly detection is larger than others. Does anyone have any idea or suggestions for me to improve the localisation capability of the model?

**Comment by kikyostudypython** [comment]
hello，Have you solved the problem？

**Comment by abc-125** [comment]
Hello, 

> the model is giving a patch of area around it as anomalous

This is an expected behavior for PatchCore, even for the MVTec AD dataset, see examples [here](https://github.com/openvinotoolkit/anomalib/tree/main/src/anomalib/models/patchcore#sample-results).
I can suggest trying EfficientAD instead, it shows fewer false positives on the anomaly map than PatchCore. Or you can look into recent supervised anomaly detection models, [PRN](https://openaccess.thecvf.com/content/CVPR2023/papers/Zhang_Prototypical_Residual_Networks_for_Anomaly_Detection_and_Localization_CVPR_2023_paper.pdf) promises to have more accurate localization of defects than unsupervised models.
